### PR TITLE
Recruiter Form Feature 

### PR DESCRIPTION
**Pull request by Ankita2002-Fr**
Hello,

The recruiter data is being added in Firebase.

Result contains the following data: 

Message: Request Submission Successful!!
Recruiter Id
Recruiter Name
Member Name
Timestamp

Please verify the changes and let me know your comments.

### CODE REVIEW

File: models/recruiters.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    logger.error('Error in adding recruiter', err)
+    throw err
+  }    
+}
+
+    
+const sendDetails= async(recruiter, username, timeStamp) =>{
+  try{
+    const user = await userModel.where('username', '==', username).limit(1).get()
+    var userName;
```
Comment (PremanandChowdhury) [inline models/recruiters.js:31]: Please address the warnings here.

File: controllers/recruitersController.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const recruiterQuery = require('../models/recruiters')
+
+
+const addRecruiter = async (req, res) => {
+  try {
+    const timeStamp=req._startTime
+    const result = await recruiterQuery.addRecruiterInfo(req.body, req.params.username, timeStamp)
+    res.send(result)
```
Comment (ankurnarkhede) [inline controllers/recruitersController.js:8]: this should be `return res.send....`

File: models/recruiters.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    if(!user.empty)
+    {
+      user.forEach(doc => {
+        const userFname = doc.data().first_name
```
Comment (PremanandChowdhury) [inline models/recruiters.js:35]: Use ``camelCase`` instead of ``snake_case``
The consistency while writting the code is important.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]
@Ankita2002-Fr , can yo address the errors, warnings and add tests to the routes?

File: models/recruiters.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        const userLname = doc.data().last_name
+        userName = userFname+" "+userLname
```
Comment (PremanandChowdhury) [inline models/recruiters.js:37]: Use ``userFirstName`` instead of ``userFname``
&     ``userLastName`` instead of ``userLname``

File: controllers/recruitersController.js (hunk 2 above)
Comment (swarajpure) [reply controllers/recruitersController.js:8]: `return res.json....`
right? @ankurnarkhede

Comment (ankurnarkhede) [reply controllers/recruitersController.js:8]: Yes, @swarajpure

File: routes/members.js [hunk 5]
```diff
@@ -50,4 +50,6 @@ router.get('/', membersController.getMembers)
 
 router.get('/idle', membersController.getIdleMembers)
 
+router.post('/intro/:username', recruitersController.addRecruiter)
```
Comment (ankurnarkhede) [inline routes/members.js:53]: Can you add the api definition here and generate the schema?
Refer to the other routes and https://github.com/Real-Dev-Squad/website-backend/blob/develop/README.md#api-documentation

File: controllers/recruitersController.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const recruiterQuery = require('../models/recruiters')
+
+const addRecruiter = async (req, res) => {
+  try {
+    const result = await recruiterQuery.addRecruiterInfo(req)
```
Comment (ankurnarkhede) [inline controllers/recruitersController.js:5]: We should pass req.body.
Also, we should validate the data.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: controllers/recruitersController.js (hunk 6 above)
Comment (Ankita2002-Fr) [reply controllers/recruitersController.js:5]: We need to access timestamp so 'req' has been passed over there.

File: models/recruiters.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        const userEmail = doc.data().email
+        userName = userFirstName + ' ' + userLastName + ' (' + userEmail + ')'
+      })
+    }
+    return {
+      message: 'Request Submission Successful!!',
+      id: recruiterInfo.id,
+      recruiterName: recruiter.data().firstName + ' ' + recruiter.data().lastName,
+      userName: userName,
+      timestamp: req._startTime
```
Comment (ankurnarkhede) [inline models/recruiters.js:31]: What is this timestamp meant for? Can we use Date.now() here?

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: models/recruiters.js (hunk 7 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:31]: I have replaced it now by new Date().toUTCString().

File: controllers/recruitersController.js (hunk 6 above)
Comment (Ankita2002-Fr) [reply controllers/recruitersController.js:5]: I have used req.body and req.params.username instead of req.

File: routes/members.js (hunk 5 above)
Comment (Ankita2002-Fr) [reply routes/members.js:53]: Added the api definition and generated the schema successfully.

File: models/recruiters.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+const addRecruiterInfo = async (recruiterData, username) => {
+  try {
+    // Add the recruiter data in DB
+    const recruiterInfo = await recruiterModel.add(recruiterData)
+    // Fetch the recruiter from DB
+    const recruiter = await recruiterModel.doc(recruiterInfo.id).get()
+    // Fetch the user from DB
+    const user = await userModel.where('username', '==', username).limit(1).get()
```
Comment (ankurnarkhede) [inline models/recruiters.js:25]: @Ankita2002-Fr , we can use the fetchUser function here: https://github.com/Real-Dev-Squad/website-backend/blob/develop/models/users.js#L88

File: models/recruiters.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        const userEmail = doc.data().email
+        userName = userFirstName + ' ' + userLastName + ' (' + userEmail + ')'
+      })
+    }
+    return {
+      message: 'Request Submission Successful!!',
+      id: recruiterInfo.id,
+      recruiterName: recruiter.data().first_name + ' ' + recruiter.data().last_name,
+      userName: userName,
+      timestamp: new Date().toUTCString()
```
Comment (ankurnarkhede) [inline models/recruiters.js:40]: We should store the unix timestamp in the DB.
Simple use Date.now() here, the consumer can modify it in the way they want later.

File: package-lock.json [hunk 10]
```diff
@@ -1,8543 +1,8 @@
 {
```
Comment (ankurnarkhede) [inline package-lock.json:1]: There looks no change in package.json, so package-lock should not be modified.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: models/recruiters.js (hunk 9 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:40]: The field "timestamp" is not a part of recruiter model.
 Do you want me to store timestamp in DB?

File: models/recruiters.js (hunk 8 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:25]: I think, the fetchUser function is changing the username as null.
So I am not able to fetch the data for null username.
Can you please assist me on this.

Please check the screenshot below.

Capture

Comment (Ankita2002-Fr) [reply models/recruiters.js:25]: No worry, I have figured it out!

File: models/recruiters.js (hunk 9 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:40]: No worry, I have figured it out!

File: models/recruiters.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    // Fetch the recruiter from DB
+    const recruiter = await recruiterModel.doc(recruiterInfo.id).get()
+    // Fetch the user from DB
+    const user = await userModel.where('username', '==', username).limit(1).get()
+    let userName
+    if (!user.empty) {
+      user.forEach(doc => {
+        const userFirstName = doc.data().first_name
+        const userLastName = doc.data().last_name
+        const userEmail = doc.data().email
```
Comment (swarajpure) [inline models/recruiters.js:31]: We could use object destructuring https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring

Also, if you want to rename them while destructuing, you can do that. This will help: https://wesbos.com/destructuring-renaming

`const { first_name: userFirstName, last_name: userLastName, email:userEmail } = doc.data()`

File: routes/members.js [hunk 12]
```diff
@@ -50,4 +52,29 @@ router.get('/', membersController.getMembers)
 
 router.get('/idle', membersController.getIdleMembers)
 
+/**
+ * @swagger
+ * /members/intro/{username}:
```
Comment (swarajpure) [inline routes/members.js:57]: `/members/intro/:username`

File: routes/members.js [hunk 13]
```diff
@@ -50,4 +52,29 @@ router.get('/', membersController.getMembers)
...
+ *     summary: Posts details of the recruiter
+ *     tags:
+ *       - Members
+ *     responses:
+ *       200:
+ *         description: Details of the recruiter and the member in which recruiter is interested
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/recruiters'
```
Comment (swarajpure) [inline routes/members.js:68]: I can't find this schema 🤔 
I guess that's remaining to add

**Review by swarajpure — CHANGES_REQUESTED** [review]
Great job! ⚡ 

Some changes required. Please check.

File: middlewares/validators/recruiter.js [hunk 14]
```diff
@@ -9,7 +9,8 @@ const validateRecruiter = async (req, res, next) => {
     reason: joi.string().optional(),
     email: joi.string().optional(),
     currency: joi.string().optional(),
-    package: joi.number().optional()
+    package: joi.number().optional(),
+    timestamp: joi.string().optional()
```
Comment (swarajpure) [inline middlewares/validators/recruiter.js:13]: timestamp should be `number`, right?

**Comment by Ankita2002-Fr** [comment]
I am attaching the screenshots of response body and Firestore.
Please check!

Postman

Firestore-recruiter

File: routes/members.js (hunk 13 above)
Comment (ankurnarkhede) [reply routes/members.js:68]: @Ankita2002-Fr

File: test/integration/recruiterController.test.js [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+const cleanDb = require('../utils/cleanDb')
+
+chai.use(chaiHttp)
+
+describe('Recruiters', function () {
+  let username
+  beforeEach(async function () {
+    const userId = await addUser()
+    const { user } = await users.fetchUser({ userId })
+    username = user.username
```
Comment (swarajpure) [inline test/integration/recruiterController.test.js:17]: We can destructure the username here itself
`const { user: { username }} = await users.fetchUser({ userId })`

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: controllers/recruitersController.js [hunk 16]
```diff
@@ -13,7 +13,7 @@ const addRecruiter = async (req, res) => {
     return res.json(result)
   } catch (error) {
     logger.error(`Error while adding recruiterInfo: ${error}`)
-    return res.boom.serverUnavailable('Something went wrong please contact admin')
+    return res.boom.notFound('Something went wrong please contact admin')
```
Comment (swarajpure) [inline controllers/recruitersController.js:16]: `notFound`? 🤔

**Review by swarajpure — COMMENTED** [review]
Some minor things... and a query
Please check 😅

File: test/integration/recruiterController.test.js (hunk 15 above)
Comment (Ankita2002-Fr) [reply test/integration/recruiterController.test.js:17]: code1


If I destructure in beforeEach then I am not able to access the username outside this block.
If I declare the username globally then it is  redeclaring in beforeEach and  getting  error.

File: controllers/recruitersController.js (hunk 16 above)
Comment (Ankita2002-Fr) [reply controllers/recruitersController.js:16]: Should I change it back to serverUnavailable?

File: test/integration/recruiterController.test.js (hunk 15 above)
Comment (swarajpure) [reply test/integration/recruiterController.test.js:17]: I see. Becoming JS level issue now.
What you could do is declare it as `let username` just before the `beforeEach` inside the `Recruiters` describe block. Don't assign it value just yet. This will allow the whole `Recruiters` describe block, to access the variable `username`.

Then at the time of destructuring you could assign value to it. Remove `const` while destructuring.

Hopefully this should work

File: controllers/recruitersController.js (hunk 16 above)
Comment (swarajpure) [reply controllers/recruitersController.js:16]: Yes. If you want to handle the case of concerned user not found, that has to be handled differently. Please check `makeNewBid` function in the `auctionController.js` to get a fair idea of how to handle such things. Also see the concerned model function.

File: middlewares/validators/recruiter.js [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+  const schema = joi.object().keys({
+    company: joi.string().optional(),
+    first_name: joi.string().optional(),
+    last_name: joi.string().optional(),
+    designation: joi.string().optional(),
+    reason: joi.string().optional(),
+    email: joi.string().optional(),
+    currency: joi.string().optional(),
+    package: joi.number().optional(),
+    timestamp: joi.string().optional()
```
Comment (Rucha1499) [inline middlewares/validators/recruiter.js:13]: I have a doubt regarding these fields. Why are we having all of them optional ?

Comment (Rucha1499) [inline middlewares/validators/recruiter.js:13]: Are we getting `timestamp` from frontend? Backend is generating that right? Please correct if I am wrong. 😃

Comment (PremanandChowdhury) [reply middlewares/validators/recruiter.js:13]: It is like we need the time at which the Recruiter Contacted a member.
This will help us to add in the `Notification` of member.

Comment (PremanandChowdhury) [reply middlewares/validators/recruiter.js:13]: - Every field except the package should be `required`
- Because other fields are important and as per the [Issue](https://github.com/Real-Dev-Squad/website-members/issues/127 "Recruiter Issue")

Comment (swarajpure) [reply middlewares/validators/recruiter.js:13]: What she meant was, do we really need to validate on `timestamp`? It won't be coming in from `req.body`, right?

We should NOT be sending timestamp from frontend (if we are, just in case). Any user could send random timestamp (of future or past) and that would get stored in db, which is not something we want.

Comment (Ankita2002-Fr) [reply middlewares/validators/recruiter.js:13]: @Rucha1499 Sorry it is added mistakenly.
It is not coming through req (from frontend) ,we are later adding it in the DB.

Comment (PremanandChowdhury) [reply middlewares/validators/recruiter.js]: > What she meant was, do we really need to validate on `timestamp`? It won't be coming in from `req.body`, right?
> 
> We should NOT be sending timestamp from frontend (if we are, just in case). Any user could send random timestamp (of future or past) and that would get stored in db, which is not something we want.

My Bad 😅

**Comment by PremanandChowdhury** [comment]
@Ankita2002-Fr Please resolve the conflicts and push.

**Comment by Ankita2002-Fr** [comment]
> @Ankita2002-Fr , can yo address the errors, warnings and add tests to the routes?

Done!

File: controllers/recruitersController.js (hunk 2 above)
Comment (Ankita2002-Fr) [reply controllers/recruitersController.js]: Done!

File: models/recruiters.js (hunk 11 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js]: Done!

File: routes/members.js (hunk 13 above)
Comment (Ankita2002-Fr) [reply routes/members.js:68]: Done!

File: routes/members.js (hunk 12 above)
Comment (Ankita2002-Fr) [reply routes/members.js]: Done!

File: test/integration/recruiterController.test.js [hunk 18]
```diff
@@ -0,0 +1,53 @@
+const chai = require('chai')
+const { expect } = chai
+const chaiHttp = require('chai-http')
+
```
Comment (swarajpure) [inline test/integration/recruiterController.test.js:4]: Please remove the word `controller` from the filename

File: controllers/recruitersController.js [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+ * @param res {Object} - Express response object
+ */
+
+const addRecruiter = async (req, res) => {
+  try {
+    const result = await recruiterQuery.addRecruiterInfo(req.body, req.params.username)
+    return res.json(result)
+  } catch (error) {
+    logger.error(`Error while adding recruiterInfo: ${error}`)
+    return res.boom.serverUnavailable('Something went wrong please contact admin')
```
Comment (swarajpure) [inline controllers/recruitersController.js:16]: We should be handling the case when the given username does not exist.
Return 404 in that case using `res.boom.notFound...`

Check this out for more info: https://github.com/Real-Dev-Squad/website-backend/blob/7bc36f4efe28d434f9f55bcbdab10c66bb1f7aed/controllers/auction.js#L73

The model function will return a variable if the user is not found and we can set error code based on it in controller

File: routes/members.js [hunk 20]
```diff
@@ -50,4 +52,29 @@ router.get('/', membersController.getMembers)
...
+ *
+ *       500:
+ *         description: serverUnavailable
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/components/schemas/errors/serverUnavailable'
+ */
```
Comment (swarajpure) [inline routes/members.js:76]: 404 missing in case of user not found

File: controllers/recruitersController.js [hunk 21]
```diff
@@ -0,0 +1,22 @@
+const recruiterQuery = require('../models/recruiters')
+
+/**
```
Comment (swarajpure) [inline controllers/recruitersController.js:3]: Please remove the word `controller` from the filename
Check this out: https://github.com/Real-Dev-Squad/website-backend/issues/280

**Review by swarajpure — CHANGES_REQUESTED** [review]
Requesting some changes for handling edge cases, please check

File: test/integration/recruiterController.test.js (hunk 18 above)
Comment (Ankita2002-Fr) [reply test/integration/recruiterController.test.js]: Done!

File: controllers/recruitersController.js (hunk 19 above)
Comment (Ankita2002-Fr) [reply controllers/recruitersController.js]: Handled the error for user.

File: routes/members.js (hunk 20 above)
Comment (Ankita2002-Fr) [reply routes/members.js]: Added the 404 for user not found.

File: controllers/recruitersController.js (hunk 21 above)
Comment (Ankita2002-Fr) [reply controllers/recruitersController.js]: Done!

**Review by swarajpure — APPROVED** [review]
Looks good to me. Technical part looks good.

Let the FO @PremanandChowdhury have a look at the functioning part and then we can merge this. Please do not merge before his approval.

Good work here! @Ankita2002-Fr and @ASHWINIAJIT 🚀 🔥

File: models/recruiters.js [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+    if (!userExists) {
+      return userExists
+    }
+    recruiterData.timestamp = Date.now()
+    // Add the recruiter data in DB
+    const recruiterInfo = await recruiterModel.add(recruiterData)
+    // Fetch the recruiter from DB
+    const recruiter = (await recruiterModel.doc(recruiterInfo.id).get()).data()
+    return {
+      message: 'Request Submission Successful!!',
```
Comment (ankurnarkhede) [inline models/recruiters.js:32]: Model functions should just return the data from the database. It shouldn't perform the response message generation. It can be done in the controller.

File: routes/members.js [hunk 23]
```diff
@@ -50,4 +52,36 @@ router.get('/', membersController.getMembers)
...
+
+router.post('/intro/:username', recruiterValidator.validateRecruiter, recruitersController.addRecruiter)
```
Comment (ankurnarkhede) [inline routes/members.js:85]: Why does the route need the `username` query param? 🤔
Also, is this route supposed to be consumed by anonymous users too?

**Review by ankurnarkhede — DISMISSED** [review]

File: routes/members.js (hunk 23 above)
Comment (PremanandChowdhury) [reply routes/members.js:85]: The form is linked with the members of RDS, that's the reason we need the `username` this is equivalent to RDS member username. 
Anonymous users( Recruiters: Genuine Case ) will trigger this route.

File: models/recruiters.js (hunk 22 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js]: Response message added in controller.

File: test/integration/recruiters.test.js [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+        .send({
+          company: 'Test-feature',
+          first_name: 'Ankita',
+          last_name: 'Bannore',
+          designation: 'Student',
+          reason: 'Test',
+          email: 'abc@gmail.com',
+          currency: '$',
+          package: 100000
+        })
```
Comment (swarajpure) [inline test/integration/recruiters.test.js:38]: Why not use the data from fixtures?

File: test/integration/recruiters.test.js [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
```
Comment (swarajpure) [inline test/integration/recruiters.test.js:66]: re: fixtures

File: models/recruiters.js [hunk 26]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+const addRecruiterInfo = async (recruiterData, username) => {
+  try {
+    // Fetch the user from DB
+    const { userExists, user: { first_name: userFirstName, last_name: userLastName, email: userEmail } } = await userModel.fetchUser({ username })
+    if (!userExists) {
+      return userExists
+    }
+    const userInfo = userFirstName + ' ' + userLastName + ' (' + userEmail + ')'
```
Comment (swarajpure) [inline models/recruiters.js:25]: Using template literals would make it more readable, what do you say?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: models/recruiters.js [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+    recruiterData.timestamp = Date.now()
+    // Add the recruiter data in DB
+    const recruiterInfo = await recruiterModel.add(recruiterData)
+    // Fetch the recruiter from DB
+    const recruiter = (await recruiterModel.doc(recruiterInfo.id).get()).data()
+    return {
+      recruiterId: recruiterInfo.id,
+      recruiterName: recruiter.first_name + ' ' + recruiter.last_name,
```
Comment (swarajpure) [inline models/recruiters.js:33]: re: template literals

File: routes/members.js (hunk 13 above)
Comment (swarajpure) [reply routes/members.js:68]: still can't find it
you've to add it in `docs/swaggerDefinition.js`

File: test/integration/recruiters.test.js [hunk 28]
```diff
@@ -0,0 +1,82 @@
...
+
+  afterEach(async function () {
+    await cleanDb()
+  })
+
+  describe('POST /members/intro/:username', function () {
+    it('Should return success response after adding recruiter data', function (done) {
+      chai
+        .request(app)
+        .post('/members/intro/' + username)
```
Comment (swarajpure) [inline test/integration/recruiters.test.js:28]: re: template literals

**Review by swarajpure — CHANGES_REQUESTED** [review]
Some changes requested:
- Please add schema
- Using template literals instead of string concatenation would more it easier to read in certain places

Thank you! 😄

File: test/integration/recruiters.test.js (hunk 24 above)
Comment (Ankita2002-Fr) [reply test/integration/recruiters.test.js:38]: Done!

File: test/integration/recruiters.test.js (hunk 25 above)
Comment (Ankita2002-Fr) [reply test/integration/recruiters.test.js:66]: Added recruiter data from fixture instead adding manually.

File: models/recruiters.js (hunk 26 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:25]: Used template literals.

File: models/recruiters.js (hunk 27 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:33]: Used template literals.

File: test/integration/recruiters.test.js (hunk 28 above)
Comment (Ankita2002-Fr) [reply test/integration/recruiters.test.js:28]: Added recruiters under schema.

Used template literals.

File: controllers/recruiters.js [hunk 29]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+const addRecruiter = async (req, res) => {
+  try {
+    const result = await recruiterQuery.addRecruiterInfo(req.body, req.params.username)
+    if (!result) {
+      return res.boom.notFound('User doesn\'t exist')
+    }
+    result.message = 'Request Submission Successful!!'
+    return res.json(result)
```
Comment (swarajpure) [inline controllers/recruiters.js:17]: Doesn't this look good?
```JS
return res.json({
  message: 'Request Submission Successful!!',
  result
})
```

File: models/recruiters.js [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+  try {
+    // Fetch the user from DB
+    const { userExists, user: { first_name: userFirstName, last_name: userLastName, email: userEmail } } = await userModel.fetchUser({ username })
+    if (!userExists) {
+      return userExists
+    }
+    const userInfo = `${userFirstName} ${userLastName} (${userEmail})`
+    recruiterData.timestamp = Date.now()
+    // Add the recruiter data in DB
+    const recruiterInfo = await recruiterModel.add(recruiterData)
```
Comment (sumitd94) [inline models/recruiters.js:28]: NIT: since from `recruiterInfo` we are only using the `id`, we can destructure this to something like this

```javascript
    const { id } = await recruiterModel.add(recruiterData)
```

And use this `id` directly on line no 30 and 32

File: models/recruiters.js [hunk 31]
```diff
@@ -0,0 +1,45 @@
...
+    // Fetch the recruiter from DB
+    const recruiter = (await recruiterModel.doc(recruiterInfo.id).get()).data()
```
Comment (sumitd94) [inline models/recruiters.js:30]: NIT: We can destructure this `recruiter` object, so that it makes it more readable.

File: routes/members.js [hunk 32]
```diff
@@ -1,6 +1,8 @@
 const express = require('express')
 const router = express.Router()
 const members = require('../controllers/members')
+const recruitersController = require('../controllers/recruiters')
+const recruiterValidator = require('../middlewares/validators/recruiter')
```
Comment (sumitd94) [inline routes/members.js:5]: NIT: If I was you, I would have destructured these two as well.

```javascript
const { addRecruiter } = require('../controllers/recruiters')
const { validateRecruiter } = require('../middlewares/validators/recruiter')
```
And use this value on line no 85

**Review by sumitd94 — DISMISSED** [review]
Few NIT Comments, Looks great. Good job 🥇

File: models/recruiters.js (hunk 30 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:28]: I have destructured recruiterInfo as you suggested.

File: routes/members.js (hunk 32 above)
Comment (Ankita2002-Fr) [reply routes/members.js:5]: I have destructured above as you suggested.

File: models/recruiters.js (hunk 31 above)
Comment (Ankita2002-Fr) [reply models/recruiters.js:30]: I have destructured recruiter object as you suggested.

**Review by PremanandChowdhury — APPROVED** [review]
Few comments to resolve and the PR is good to be merged for production.
Good Job 🥳

File: controllers/recruiters.js (hunk 29 above)
Comment (Ankita2002-Fr) [reply controllers/recruiters.js]: Modified the response object as you suggested.

**Review by swarajpure — APPROVED** [review]
Looks good to me!

Great job there @Ankita2002-Fr @ASHWINIAJIT 🥳

**Review by sumitd94 — APPROVED** [review]
Excellent work 🥇  @Ankita2002-Fr @ASHWINIAJIT

**Review by PremanandChowdhury — APPROVED** [review]
Great work 🎉 
Keep up the good work, many more to come 👍🏼 
@Ankita2002-Fr  & @ASHWINIAJIT
